pkg/virt-handler/vm.go

```diff
@@ -519,127 +515,77 @@ func MapPersistentVolumes(vm *v1.VirtualMachine, clientset kubecli.KubevirtClien
 			}
 
 			logger.Infof("Mapping PVC %s", pv.Name)
-			newDisk, err := mapPVToDisk(&disk, pv)
-
-			if err != nil {
+			if err = mapPVToDisk(&volume, pv); err != nil {
 				logger.Reason(err).Errorf("Mapping PVC %s failed", pv.Name)
 				return vm, err
 			}
+		}
 
-			vmCopy.Spec.Domain.Devices.Disks[idx] = *newDisk
-		} else if disk.Type == "network" {
-			newDisk := v1.Disk{}
-			model.Copy(&newDisk, disk)
-
-			if disk.Source.Host == nil {
-				logger.Error("Missing disk source host")
-				return vm, fmt.Errorf("Missing disk source host")
+		// After a PVC translation, ISCSI can be the resolved type, so "if" instead of "else if"
+		if volume.ISCSI != nil {
+			// FIXME ugly hack to resolve the IP from dns, since qemu is not in the right namespace
+			hostPort := strings.Split(volume.ISCSI.TargetPortal, ":")
+			ipAddrs, err := net.LookupIP(hostPort[0])
+			if err != nil || len(ipAddrs) < 1 {
+				return nil, fmt.Errorf("unable to resolve host '%s': %s", hostPort[0], err)
 			}
-
-			ipAddrs, err := net.LookupIP(disk.Source.Host.Name)
-			if err != nil || ipAddrs == nil || len(ipAddrs) < 1 {
-				logger.Reason(err).Errorf("Unable to resolve host '%s'", disk.Source.Host.Name)
-				return vm, fmt.Errorf("Unable to resolve host '%s': %s", disk.Source.Host.Name, err)
+			volume.ISCSI.TargetPortal = ipAddrs[0].String()
+			for _, part := range hostPort[1:] {
+				volume.ISCSI.TargetPortal = volume.ISCSI.TargetPortal + ":" + part
 			}
-
-			newDisk.Source.Host.Name = ipAddrs[0].String()
-
-			vmCopy.Spec.Domain.Devices.Disks[idx] = newDisk
 		}
+
+		// Set the translated volume, necessary since the VolumeSource might have been exchanged
+		vmCopy.Spec.Volumes[idx] = volume
 	}
 
 	return vmCopy, nil
 }
 
-func mapPVToDisk(disk *v1.Disk, pv *k8sv1.PersistentVolume) (*v1.Disk, error) {
+func mapPVToDisk(volume *v1.Volume, pv *k8sv1.PersistentVolume) error {
```

Comment:
I'm in favor of removing this PV->ISCSI mapping logic entirely at this point. This isn't a supportable use case.

We don't have to do that for this patch series or anything, but I'm curious what your thoughts are on that.

Reply:
I agree, once we can consume the pod block devices, we should remove it.

Reply:
We actually have to drop it, because qemu will not be able to access the network anymore.

However, we should keep in mind that we probably want to allow this again in future.
I.e. I can imagine that there is a flag on volumes to indicate if it should be connected through the kubelet or qemu.

Reply:
yes, but right now we don't have qemu at the right spot, thus it is still there.